File: elixir/easy/1897/1897.exs
defmodule Solution do
  @spec make_equal(words :: [String.t()]) :: boolean
  def make_equal(words) do
    counter =
      Enum.map(words, &String.to_charlist/1)
      |> Enum.concat()
      |> Enum.frequencies()

    Enum.map(counter, fn {_, v} -> rem(v, length(words)) != 0 end)
    |> Enum.any?()
    |> Kernel.!()
  end
end

ExUnit.start()

defmodule AssertionTest do
  use ExUnit.Case, async: true

  test "test_1" do
    assert Solution.make_equal(["abc", "aabc", "bc"]) == true
  end

  test "test_2" do
    assert Solution.make_equal(["ab", "a"]) == false
  end
end

File: py/easy/1897/1897.py
from collections import Counter
from typing import List


class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        counter = Counter()
        for word in words:
            counter.update(word)

        len_words = len(words)
        for value in counter.values():
            if value % len_words != 0:
                return False
        return True


solution = Solution()
assert solution.makeEqual(words=["abc", "aabc", "bc"]) is True
assert solution.makeEqual(words=["ab", "a"]) is False
